tools/generate-examples.rb

```diff
@@ -0,0 +1,409 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require "fileutils"
+require "pathname"
+
```

Comment:
It's a small nit but you could make a lot of strings below less complex with a constant like
```
EXAMPLE_DIR = File.expand_path("../examples", __dir__)
```